sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -1399,6 +1399,20 @@ func (host *pythonLanguageHost) RunPlugin(
 		return err
 	}
 
+	venvStdout, venvStderr, err := host.createEngineWriters(server.Context())
+	if err != nil {
+		return err
+	}
+	if err := tc.EnsureVenv(server.Context(),
```

Comment:
This only creates the venv if it does not exist, but it does not sync it if `requirements.txt` has changed